peerconnection.go

```diff
@@ -1002,51 +1153,25 @@ func (pc *PeerConnection) openSRTP() {
 				continue
 			}
 
+			delete(incomingSSRCes, ssrc)
 			localTransceivers = append(localTransceivers[:i], localTransceivers[i+1:]...)
-			go func(ssrc uint32, receiver *RTPReceiver) {
-				err := receiver.Receive(RTPReceiveParameters{
-					Encodings: RTPDecodingParameters{
-						RTPCodingParameters{SSRC: ssrc},
-					}})
-				if err != nil {
-					pc.log.Warnf("RTPReceiver Receive failed %s", err)
-					return
-				}
-
-				if err = receiver.Track().determinePayloadType(); err != nil {
-					pc.log.Warnf("Could not determine PayloadType for SSRC %d", receiver.Track().SSRC())
-					return
-				}
-
-				pc.mu.RLock()
-				defer pc.mu.RUnlock()
-
-				sdpCodec, err := pc.currentLocalDescription.parsed.GetCodecForPayloadType(receiver.Track().PayloadType())
-				if err != nil {
-					pc.log.Warnf("no codec could be found in RemoteDescription for payloadType %d", receiver.Track().PayloadType())
-					return
-				}
-
-				codec, err := pc.api.mediaEngine.getCodecSDP(sdpCodec)
-				if err != nil {
-					pc.log.Warnf("codec %s in not registered", sdpCodec)
-					return
-				}
-
-				receiver.Track().mu.Lock()
-				receiver.Track().kind = codec.Type
-				receiver.Track().codec = codec
-				receiver.Track().mu.Unlock()
-
-				if pc.onTrackHandler != nil {
-					pc.onTrack(receiver.Track(), receiver)
-				} else {
-					pc.log.Warnf("OnTrack unset, unable to handle incoming media streams")
-				}
-			}(ssrc, t.Receiver)
+			go startReceiver(ssrc, t.Receiver)
 			break
 		}
 	}
+
+	if remoteIsPlanB {
```

Comment:
Feel free to push back on this (might be a nit)

Do you think it is worth creating the `incomingSSRCes` at the very beginning of `SetRemoteDescription` ? It would be nice to convert some of those warnings into errors we can return back to the user. Then just pass the slice into `openSRTP`

IMO this LGTM merge wise! I would rather unblock you, and we can move forward. I should have time to test more tomorrow and since this isn't the default path I am not worried about users hitting weird cases at all!